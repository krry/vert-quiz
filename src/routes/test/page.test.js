import { cleanup, render, fireEvent, screen } from '@testing-library/svelte';
import { afterEach, describe, expect, it } from 'vitest';
import Test from './+page.svelte';

describe('Introversion Test', () => {
	afterEach(() => cleanup());
	it('page mounts', () => {
		const { container } = render(Test);
		expect(container).toBeTruthy();
		expect(container.innerHTML).toContain('parties');
		expect(container).toMatchSnapshot();
	});

	it('test submits', async () => {
		render(Test, {
			form: { taken: false }
		});
		fireEvent.submit(screen.getByRole('form'));
	});

	it('5 scores extroverted', async () => {
		render(Test, {
			answer1: 1,
			answer2: 1,
			answer3: 1,
			answer4: 1,
			answer5: 1
		});
		fireEvent.submit(screen.getByRole('form'));
		expect(screen.findByText('extroverted'));
	});

	it('4 scores extroverted', async () => {
		render(Test, {
			answer1: 1,
			answer2: 1,
			answer3: 1,
			answer4: 1,
			answer5: 1
		});
		fireEvent.submit(screen.getByRole('form'));
		expect(screen.findByText('extroverted'));
	});

	it('3 scores extroverted', () => {
		render(Test, {
			answer1: 1,
			answer2: 0,
			answer3: 1,
			answer4: 0,
			answer5: 1
		});
		fireEvent.submit(screen.getByRole('form'));
		expect(screen.findByText('extroverted'));
	});

	it('2 scores introverted', () => {
		render(Test, {
			answer1: 0,
			answer2: 0,
			answer3: 0,
			answer4: 1,
			answer5: 1
		});
		fireEvent.submit(screen.getByRole('form'));
		expect(screen.findByText('introverted'));
	});

	it('1 scores introverted', () => {
		render(Test, {
			answer1: 0,
			answer2: 0,
			answer3: 0,
			answer4: 1,
			answer5: 0
		});
		fireEvent.submit(screen.getByRole('form'));
		expect(screen.findByText('introverted'));
	});

	it('0 scores introverted', () => {
		render(Test, {
			answer1: 0,
			answer2: 0,
			answer3: 0,
			answer4: 0,
			answer5: 0
		});
		fireEvent.submit(screen.getByRole('form'));
		expect(screen.findByText('introverted'));
	});
});
