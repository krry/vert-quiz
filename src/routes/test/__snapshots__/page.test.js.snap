// Vitest Snapshot v1

exports[`Introversion Test > page mounts 1`] = `
<body>
  <div>
    <h2>
      Introversion/Extroversion Test
    </h2>
     
    <form
      method="post"
      role="form"
    >
      <label
        for="question1"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like parties?
        </h3>
         
        <select
          id="question1"
          name="question1"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question2"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like singing with friends?
        </h3>
         
        <select
          id="question2"
          name="question2"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question3"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like dancing in public?
        </h3>
         
        <select
          id="question3"
          name="question3"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question4"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like talking with groups of friends?
        </h3>
         
        <select
          id="question4"
          name="question4"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question5"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you prefer solo or multiplayer games?
        </h3>
         
        <select
          id="question5"
          name="question5"
        >
          <option
            value="0"
          >
            Solo
          </option>
          <option
            value="1"
          >
            Multiplayer
          </option>
        </select>
      </label>
       
      <div
        class="button-case svelte-ua52vp"
      >
        <button
          name="test-submit"
          type="submit"
        >
          Submit Answers
        </button>
      </div>
    </form>
    
  </div>
</body>
`;

exports[`Personality Test > page mounts 1`] = `
<body>
  <div>
    <h2>
      Introversion/Extroversion Test
    </h2>
     
    <form
      method="post"
      role="form"
    >
      <label
        for="question1"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like parties?
        </h3>
         
        <select
          id="question1"
          name="question1"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question2"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like singing with friends?
        </h3>
         
        <select
          id="question2"
          name="question2"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question3"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like dancing in public?
        </h3>
         
        <select
          id="question3"
          name="question3"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question4"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like talking with groups of friends?
        </h3>
         
        <select
          id="question4"
          name="question4"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question5"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you prefer solo or multiplayer games?
        </h3>
         
        <select
          id="question5"
          name="question5"
        >
          <option
            value="0"
          >
            Solo
          </option>
          <option
            value="1"
          >
            Multiplayer
          </option>
        </select>
      </label>
       
      <div
        class="button-case svelte-ua52vp"
      >
        <button
          name="test-submit"
          type="submit"
        >
          Submit Answers
        </button>
      </div>
    </form>
    
  </div>
</body>
`;

exports[`Test page > mounts 1`] = `
<body>
  <div>
    <h2>
      Introversion/Extroversion Test
    </h2>
     
    <form
      method="post"
      role="form"
    >
      <label
        for="question1"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like parties?
        </h3>
         
        <select
          id="question1"
          name="question1"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question2"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like singing with friends?
        </h3>
         
        <select
          id="question2"
          name="question2"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question3"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like dancing in public?
        </h3>
         
        <select
          id="question3"
          name="question3"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question4"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like talking with groups of friends?
        </h3>
         
        <select
          id="question4"
          name="question4"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question5"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you prefer solo or multiplayer games?
        </h3>
         
        <select
          id="question5"
          name="question5"
        >
          <option
            value="0"
          >
            Solo
          </option>
          <option
            value="1"
          >
            Multiplayer
          </option>
        </select>
      </label>
       
      <div
        class="button-case svelte-ua52vp"
      >
        <button
          name="test-submit"
          type="submit"
        >
          Submit Answers
        </button>
      </div>
    </form>
    
  </div>
</body>
`;

exports[`Test page > page mounts 1`] = `
<body>
  <div>
    <h2>
      Introversion/Extroversion Test
    </h2>
     
    <form
      method="post"
      role="form"
    >
      <label
        for="question1"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like parties?
        </h3>
         
        <select
          id="question1"
          name="question1"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question2"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like singing with friends?
        </h3>
         
        <select
          id="question2"
          name="question2"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question3"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like dancing in public?
        </h3>
         
        <select
          id="question3"
          name="question3"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question4"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you like talking with groups of friends?
        </h3>
         
        <select
          id="question4"
          name="question4"
        >
          <option
            value="0"
          >
            No
          </option>
          <option
            value="1"
          >
            Yes
          </option>
        </select>
      </label>
       
      <label
        for="question5"
      >
        <h3
          class="svelte-ua52vp"
        >
          Do you prefer solo or multiplayer games?
        </h3>
         
        <select
          id="question5"
          name="question5"
        >
          <option
            value="0"
          >
            Solo
          </option>
          <option
            value="1"
          >
            Multiplayer
          </option>
        </select>
      </label>
       
      <div
        class="button-case svelte-ua52vp"
      >
        <button
          name="test-submit"
          type="submit"
        >
          Submit Answers
        </button>
      </div>
    </form>
    
  </div>
</body>
`;
